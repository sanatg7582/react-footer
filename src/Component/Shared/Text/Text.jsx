import React from "react";
import PropTypes from "prop-types";
import styles from "./Text.module.scss";
/**
 * Text function for return text
 * @param {boolean} strong
 * @param {string} heading
 */
const Text = props => {
  const { strong, heading } = props;
  const headingClass = strong ? styles.footerHeading : "";
  return <span className={headingClass}>{heading}</span>;
};
Text.defaultProps = {
  strong: false
};
// PropsType validation here
Text.propTypes = {
  heading: PropTypes.string,
  strong: PropTypes.bool
};
export default Text;
