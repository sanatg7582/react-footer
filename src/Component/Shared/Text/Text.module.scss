.footerHeading {
  font-size: 17px;
  font-weight: 500;
  padding: 3px;
}
