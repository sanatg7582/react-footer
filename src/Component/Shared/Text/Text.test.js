import React from "react";
import { shallow } from "enzyme";
import Text from "./Text";

describe("renders Text without crashing ", () => {
  const wrapper = shallow(<Text strong={true} />);
  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });
  it("Text component default return blank", () => {
    expect(wrapper.text()).toBe("");
  });
  it("Text class when strong props is true", () => {
    expect(wrapper.find("span").hasClass("footerHeading")).toEqual(true);
  });
  it("Snapshot test", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
