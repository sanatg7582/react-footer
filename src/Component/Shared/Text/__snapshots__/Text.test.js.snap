// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Text without crashing  Snapshot test 1`] = `ShallowWrapper {}`;
